test-suite/shared/src/test/require-jdk8/org/scalajs/testsuite/javalib/util/Base64Test.scala

```diff
@@ -0,0 +1,532 @@
+import org.junit.Test
+import org.scalajs.testsuite.utils.AssertThrows._
+
+class Base64Test {
+  import Base64Test._
+
+  // --------------------------------------------------------------------------
+  // ENCODERS
+  // --------------------------------------------------------------------------
+
+  @Test def encodeToString(): Unit = {
+    val results = encoders.map{case (name, in, enc) => name -> enc.encodeToString(in)}
+    assertEquals("calculated count doesn't match computed count", encodedResults.length, results.size)
+    results.zip(encodedResults).foreach{ case ((name, enc), exp) =>
+      assertEquals(s"encodeToString doesn't match for: $name", exp, enc)
+    }
+  }
+
+  @Test def encodeOneArray(): Unit = {
+    val results = encoders.map{case (name, in, enc) => name -> enc.encode(in)}
+    assertEquals("calculated count doesn't match computed count", encodedResults.length, results.size)
+    results.zip(encodedResults).foreach{ case ((name, enc), exp) =>
+      assertEquals(s"encode array doesn't match for: $name", exp, new String(enc, StandardCharsets.ISO_8859_1))
+    }
+  }
+
+  @Test def encodeTwoArrays(): Unit = {
+    encoders.zip(encodedResults).foreach{case ((name, in, enc), exp) =>
+      val dst = new Array[Byte](exp.length + 10) // array too big on purpose
+      val written = enc.encode(in, dst)
+      assertEquals(s"number of written bytes doesn't match for: $name", exp.length, written)
+      val content = dst.slice(0, written)
+      val rlt = new String(content, StandardCharsets.ISO_8859_1)
+      assertEquals(s"encode array into array doesn't match for: $name", exp, rlt)
+    }
+  }
+
+  @Test def encodeTwoArraysThrowsWithTooSmallDestination(): Unit = {
+    val in = "Man"
+    val dst = new Array[Byte](3) // too small
+    assertThrows(classOf[IllegalArgumentException], Base64.getEncoder.encode(in.getBytes, dst))
+  }
+
+  @Test def encodeByteBuffer(): Unit = {
+    encoders.zip(encodedResults).foreach{case ((name, in, enc), exp) =>
+      var rlt = enc.encode(ByteBuffer.wrap(in))
+      assertEquals(s"byte buffers don't match for: $name", exp, new String(rlt.array(), StandardCharsets.ISO_8859_1))
+      val bb = ByteBuffer.allocate(in.length + 2)
+      bb.position(2)
+      bb.mark()
+      bb.put(in)
+      bb.reset()
+      rlt = enc.encode(bb)
+      assertEquals(s"byte buffers don't match for: $name", exp, new String(rlt.array(), StandardCharsets.ISO_8859_1))
+    }
+  }
+
+  @Test def encodeOutputStream(): Unit = {
+    encoders.zip(encodedResults).foreach { case ((name, in, enc), exp) =>
+      val baos = new ByteArrayOutputStream()
+      val out = enc.wrap(baos)
+      out.write(in(0))
+      out.write(in, 1, in.length - 1)
+      out.close()
+      val rlt = new String(baos.toByteArray, StandardCharsets.ISO_8859_1)
+      assertEquals(s"output stream result doesn't match for: $name", exp, rlt)
+    }
+  }
+
+  // @Test // Fails on the JVM (bug JDK-8176379)
+  def encodeOutputStreamFailsOnJVM(): Unit = {
```

Comment:
If a test fails on the JVM, you should not disable the test. Instead, you should add the following at the beginning of the test:
```scala
assumeFalse("JDK bug JDK-8176379", Platform.executingInJVM)
```
with the following relevant imports at the top of the file:
```scala
import org.junit.Assume._

import org.scalajs.testsuite.utils.Platform
```